lib/iris/fileformats/pp.py

```diff
@@ -1378,8 +1378,19 @@ def save(self, file_handle):
 
         # Get the actual data content.
         data = self.data
-        if ma.is_masked(data):
-            data = data.filled(fill_value=self.bmdi)
+        mdi = self.bmdi
+        if np.any(data == mdi):
+            msg = ('PPField data contains unmasked points equal to the fill '
+                   "value, {}. As saved, these points will read back as "
+                   "missing data. To save these as normal values, please "
+                   "specify a 'fill_value' keyword not equal to any valid "
+                   "data points.")
+            warnings.warn(msg.format(mdi))
+        if isinstance(data, ma.MaskedArray):
+            if ma.is_masked(data):
```

Comment:
`ma.is_masked` can be used on `ndarray`s too, so the instance check isn't necessary.

Reply:
Except that it also qualifies the need for "data = data.data".
We could have..
```
            if ma.is_masked(data):
                data = data.filled(fill_value=mdi)
            elif isinstance(data, ma.MaskedArray):
                data = data.data
```
But frankly I think the existing is clearer ?
So I think it's simpler

Reply:
Oh sorry, I missed the `data = data.data`. I would question the need for that actually, but I won't quibble about it.